src/udev/udev-builtin-net_id.c

```diff
@@ -345,6 +345,23 @@ static int dev_pci_slot(sd_device *dev, struct netnames *names) {
                 if (sd_device_get_sysname(hotplug_slot_dev, &sysname) < 0)
                         continue;
 
+                /* match slot address to function_id, if available */
+                if (naming_scheme_has(NAMING_SLOT_FUNCTION_ID) &&
+                    sd_device_get_sysattr_value(hotplug_slot_dev, "function_id", &attr) >= 0) {
+                        int function_id;
+                        char str[PATH_MAX];
+
+                        if (safe_atoi(attr, &function_id) >= 0 &&
+                            snprintf_ok(str, sizeof str, "%s/%08x", slots, function_id) &&
+                            access(str, R_OK | X_OK) == 0) {
+                                hotplug_slot = function_id;
+                                domain = 0;
+                        } else {
+                                log_debug("No matching slot for function_id (%s).", attr);
+                        }
+                        break;
+                }
+
```

Comment:
Are we sure only s390 uses function_id as sysattr?

i.e. maybe there are other PCI subsystems that have the same sysattr, but use it differently? With this new patch we might break those?

Reply:
Good question, this is of course a bit hard to say since as you said previously the Kernel's interface is somewhat loosely defined. I can say that the following:
```
% git grep 'function_id' Documentation/ 
Documentation/s390/pci.rst:  - function_id
% git grep 'function_id' drivers/pci
%
```

That also includes `Documentation/ABI/testing/sysfs-bus-pci` which should contain
most attributes. Obviously we did fail to document `function_id` there and only have it in
our arch documentation. I'll fix that., but yes as far as I can tell we should be the only ones
using that name and once I add it to the appropriate documentation file anyone wanting to add it
should find our use.